### ISSUE TITLE
using docker - devcontainer fails azd up with MSBuild version 17.7.4+3ebbd7c49 for .NET

### ISSUE THREAD

**Issue by SQLDBAWithABeard** [Azure-Samples/azure-sql-db-session-recommender-v2#4]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [X ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>
open dev ocntainer on windows azd login and then azd up

### Any log messages given by the failure
> Packaging services (azd package)
>
>  (x) Failed: Packaging service functionapp
>
>ERROR: failed packaging service 'functionapp': failing invoking action 'package', dotnet >publish on project '/workspaces/azure-sql-db-session-recommender-v2->1/func/RequestHandler.csproj' failed: exit code: 1, stdout: MSBuild version 17.7.4+3ebbd7c49 >or .NET
>  Determining projects to restore...
> >/usr/share/dotnet/sdk/7.0.406/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFramewor
>kInference.targets(160,5): error NETSDK1045: The current .NET SDK does not support >targeting .NET 8.0.  Either target .NET 7.0 or lower, or use a version of the .NET SDK that >supports .NET 8.0. Download the .NET SDK from https://aka.ms/dotnet/download 
> [/workspaces/azure-sql-db-session-recommender-v2-1/func/RequestHandler.csproj]
>, stderr: 
>
>ERROR: error executing step command 'package --all': failed packaging service 'functionapp': >failing invoking action 'package', dotnet publish on project '/workspaces/azure-sql-db->session-recommender-v2-1/func/RequestHandler.csproj' failed: exit code: 1, stdout: MSBuild >version 17.7.4+3ebbd7c49 for .NET
 > Determining projects to restore...
>/usr/share/dotnet/sdk/7.0.406/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFramewor>kInference.targets(160,5): error NETSDK1045: The current .NET SDK does not support targeting >.NET 8.0.  Either target .NET 7.0 or lower, or use a version of the .NET SDK that supports .NET >8.0. Download the .NET SDK from https://aka.ms/dotnet/download [/workspaces/azure-sql->db-session-recommender-v2-1/func/RequestHandler.csproj]
>, stderr: 
>vscode ➜ /workspaces/azure-sql-db-session-recommender-v2-1 (main) $ 

### Expected/desired behavior
>

### OS and Version?
> Windows 11 23H2
### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by yorek** [comment]
Fixed with #6904590

**Comment by SQLDBAWithABeard** [comment]
Confirmed works here
